stdlib/public/core/StringBridge.swift

```diff
@@ -125,57 +125,77 @@ private var kCFStringEncodingUTF8 : _swift_shims_CFStringEncoding {
   @inline(__always) get { return 0x8000100 }
 }
 
-@_effects(readonly)
-private func _unsafeAddressOfCocoaStringClass(_ str: _CocoaString) -> UInt {
+@_effects(releasenone)
+private func _unsafeAddressOfCocoaStringClass(
+  _ str: _CocoaString) -> UInt {
   return _swift_stdlib_unsafeAddressOfClass(str)
 }
 
-internal enum _KnownCocoaString {
+@_effects(readonly)
+private func _lookUpObjCClass(_ name: String) -> UInt {
+  return name.withCString {
+    return _swift_stdlib_lookUpClass($0)
+  }
+}
+
+internal enum CocoaStringKind {
```

Comment:
This is now only used in -isEqual… maybe just move it there and inline the checks?